ceph/rbd/workflows/group.py

```diff
@@ -0,0 +1,138 @@
+        ns_verify_kw = {}
+        ns_verify_kw["pool-name"] = pool_name
+        ns_ls_out = rbd.namespace.list(**ns_verify_kw)
+        if namespace not in ns_ls_out[0]:
+            log.info(f"namespace {namespace} in {pool_name} is getting created")
+            ns_create_kw = {}
+            ns_create_kw["namespace"] = namespace
+            ns_create_kw["pool-name"] = pool_name
+            (_, ns_create_err) = rbd.namespace.create(**ns_create_kw)
+            if not ns_create_err:
+                log.info(
+                    "SUCCESS: Namespace {namespace_name} got created in pool {pool_name} "
+                )
+            else:
+                log.error(
+                    f"FAIL: Namespace {namespace} creation failed in pool {pool_name} with error {ns_create_err}"
+                )
+                return 1
+            group_kw.update({"namespace": namespace})
+
+    # create a group in the pool/[namespace]
+    group = kw.get("group", None)
+    if group is None:
+        log.error("Group name is must to create a group")
+        return 1
+    group_kw.update({"group": group})
+    (_, g_err) = rbd.group.create(**group_kw)
+    if not g_err:
+        log.info(f"SUCCESS: Group {group} got created in pool {pool_name}/{namespace} ")
+    else:
+        log.error(
+            f"FAIL: group {group} creation failed in pool {pool_name}/{namespace} with error {g_err}"
+        )
+        return 1
+
+    # verify group creation
+    group_kw.pop("group")
+    # group_kw.update({"format": "json"})
+    (gls_out, _) = rbd.group.list(**group_kw)
+    if group in gls_out:
+        log.info(f"Group creation {group} verification is successfull")
+        return 0
+    else:
+        log.error(f"Group creation {group} verification failed")
+        return 1
+
+
+def add_image_to_group_and_verify(**kw):
+    """
+    Adding image to the group
+    Args:
+        kw(dict): Key/value pairs that needs to be provided to the installer
+            Example::
+            Supported keys:
+                pool(str): pool nane where group should be created
+                namespace(str): pool|[namespace] where greoup should be created
+                group(str): group name to be created
+                image(str): image to be added to the group
+    """
+    rbd = Rbd(kw["client"])
+
+    # add image to the [pool]|[namespace]|<group>
+    pool = kw.get("pool", "rbd")
+    namespace = kw.get("namespace", None)
+    group = kw.get("group", None)
+    image = kw.get("image", None)
+    image_group_kw = {}
+    if pool is not None:
+        image_group_kw.update({"group-pool": pool})
+        image_group_kw.update({"image-pool": pool})
+    if namespace is not None:
+        image_group_kw.update({"namespace": namespace})
+    if group is not None:
+        image_group_kw.update({"group": group})
+    else:
+        log.error("Group is the must kw for adding image to the group")
```

Comment:
`kw` is the coding word, can you please use proper word like `config` for users to debug errors.

Reply:
:nit: but fixed

Reply:
please address this in line 118 as well here `log.error("Image is the must kw for adding image to the group")` also in others.